services/harmony/app/frontends/labels.ts

```diff
@@ -0,0 +1,61 @@
+// functions to support labels routes
+
+import { Response, NextFunction } from 'express';
+import HarmonyRequest from '../models/harmony-request';
+import { addLabelsToJobs, deleteLabelsFromJobs } from '../models/label';
+import db from '../util/db';
+import { isAdminUser } from '../util/edl-api';
+
+/**
+ * Express.js handler that adds one or more labels to a job `(POST /labels/{jobID}/add)`.
+ * Currently only the job owner can add labels (no admin access).
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ * @param next - The next function in the call chain
+ * @returns Resolves when the request is complete
+ */
+export async function addJobLabels(
+  req: HarmonyRequest, res: Response, next: NextFunction,
+): Promise<void> {
+  const isAdmin = await isAdminUser(req);
+
+  try {
+    await db.transaction(async (trx) => {
+      await addLabelsToJobs(trx, req.body.jobID, req.user, req.body.label, isAdmin);
```

Comment:
I wasn't sure about this, but I ended up using singular, because `label` is consistent with the `label` parameter we use with coverage and EDR requests (which was made singular to be consistent with `granuleId`, like you said). It would seem strange to me to use `labels` here, but `label` there.